### PR TITLE
[Frontend] Ambry blob metadata cache - 1

### PR DESCRIPTION
**Pull request by snalli**
This patch adds a cache to the non-blocking router. The cache saves
metadata of composite blobs.

### CODE REVIEW

File: ambry-commons/src/main/java/com/github/ambry/commons/AmbryCacheObject.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.commons;
+
+/**
+ * Implements a generic cache object for the AmbryCache
+ */
+public class AmbryCacheObject {
```
Comment (justinlin-linkedin) [inline ambry-commons/src/main/java/com/github/ambry/commons/AmbryCacheObject.java:20]: please set this to an interface.

**Comment by mikeyang923** [comment]
So this is to add the cache, but the cache is not in use yet? Will you raise another PR to plug the cache into the ambry read pipeline?

**Comment by snalli** [comment]
> So this is to add the cache, but the cache is not in use yet? Will you raise another PR to plug the cache into the ambry read pipeline?

Yes Mike, I plan for it to be 3 part PR where I'll integrate the cache into Ambry frontend. Based on some previous attempts, I realized the PR is going to be bigger than I originally thought and it'd be easier to review if I split it.

File: ambry-commons/src/main/java/com/github/ambry/commons/AmbryCacheObject.java (hunk 1 above)
Comment (snalli) [reply ambry-commons/src/main/java/com/github/ambry/commons/AmbryCacheObject.java]: Done

**Review by mikeyang923 — APPROVED** [review]
lgtm

**Comment by snalli** [comment]
> lgtm

Thanks. I accidentally re-clicked request for review after your approval. Not sure if it cancels the approval and re-sends a review request to the same reviewer.

**Review by justinlin-linkedin — APPROVED** [review]
LGTM

**Pull request merged**